### PR TITLE
feat: refactor networking util

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/priority backlog

**What this PR does / why we need it**:

Related to https://github.com/kubernetes/kubernetes/issues/76206 

Cleanup networking utils in the e2e framework.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by draveness** [comment]
/cc @andrewsykim 

I start to work on the refactor of networking utils and found out several problems.

1. `networking` pkg still needs `Failf` function not implemented in the `framework` package which will lead to import cycle problem.
2. the `shuffleNodes` function was both used in networking and nodes package.
3. I don't have a good idea on how to deal with the functions named with `Test` prefix.

Do you have thoughts on this? Thank you.

File: test/e2e/common/downward_api.go [hunk 1]
```diff
@@ -19,12 +19,13 @@ package common
...
 	"fmt"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	utilversion "k8s.io/apimachinery/pkg/util/version"
 	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/networking"
```
Comment (andrewsykim) [inline test/e2e/common/downward_api.go]: Let's import alias to `e2enet`?

File: test/e2e/common/networking.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	. "github.com/onsi/ginkgo"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/networking"
```
Comment (andrewsykim) [inline test/e2e/common/networking.go]: same here

File: test/e2e/framework/networking/config.go [hunk 3]
```diff
@@ -0,0 +1,688 @@
+package networking
```
Comment (andrewsykim) [inline test/e2e/framework/networking/config.go]: fixtures.go? :P 

Also missing licence header

Comment (draveness) [reply test/e2e/framework/networking/config.go]: Yeah, I'll fix this, please take a look at the comment above: 🤣 

> I start to work on the refactor of networking utils and found out several problems.
>
> + networking pkg still needs Failf function not implemented in the framework package which will lead to import cycle problem.
> + the shuffleNodes function was both used in networking and nodes package.
> + I don't have a good idea on how to deal with the functions named with Test prefix.
>
> Do you have thoughts on this? Thank you.

Comment (andrewsykim) [reply test/e2e/framework/networking/config.go]: > networking pkg still needs Failf function not implemented in the framework package which will lead to import cycle problem.

https://github.com/kubernetes/kubernetes/pull/77160 is currently refactoring wrapper functions like `Failf` but it doesn't move it to it's own package, just a dedicated file under `test/e2e/framework`. That will still cause import cycles here right?

Comment (draveness) [reply test/e2e/framework/networking/config.go]: Yeah, we need to move it to a sub package in framework to avoid it.

Comment (andrewsykim) [reply test/e2e/framework/networking/config.go]: Looks like the only reason we import `e2enet` in `test/e2e/framework/networking_utils.go` is for `PokeHTTP`, if we removed that import somehow would that resolve the import cycle for now?

Comment (draveness) [reply test/e2e/framework/networking/config.go]: Yeah, do you have any suggestions on where to place these functions?

Comment (andrewsykim) [reply test/e2e/framework/networking/config.go]: `test/e2e/framework/http`?

Comment (andrewsykim) [reply test/e2e/framework/networking/config.go]: Ah, that won't work cause PokeHTTP will depend on framework. Can we just leave PokeHTTP in networking_utils.go for now?

**Comment by draveness** [comment]
It seems like add `Failf` into the log package in really hard to avoid especially in the middle of the refactoring process, so I add the `Failf` function into `log` package to solve the import cycle problem.... any thoughts on this?

**Comment by andrewsykim** [comment]
> It seems like add Failf into the log package in really hard to avoid especially in the middle of the refactoring process, so I add the Failf function into log package to solve the import cycle problem.... any thoughts on this?

Seems awkward, do you think https://github.com/kubernetes/kubernetes/pull/77160 will help?

**Comment by draveness** [comment]
> > It seems like add Failf into the log package in really hard to avoid especially in the middle of the refactoring process, so I add the Failf function into log package to solve the import cycle problem.... any thoughts on this?
> 
> Seems awkward, do you think #77160 will help?

Yes, I'll come back to work on this after #77160 get merged.

**Comment by timothysc** [comment]
needs a rebase 
/assign @timothysc

**Comment by draveness** [comment]
> needs a rebase
> /assign @timothysc

I'll continue to work on this after #77155 get merged, this PR may have conflict files with that one.

**Comment by johnSchnake** [comment]
The PR mentioned above has merged and there are numerous conflicts now on this PR.

@draveness are you still planning on finishing this one out?

**Comment by draveness** [comment]
> The PR mentioned above has merged and there are numerous conflicts now on this PR.
> 
> @draveness are you still planning on finishing this one out?

@johnSchnake  Thanks for the reminder, please feel free to take it over if you would like to work on this one!

**Comment by johnSchnake** [comment]
@draveness I'm trying to wrap my head around what can actually be moved into its own package and I wondered if you had any tips to pass on from your analysis?

The main issue is that the `NetworkingTestConfig` wraps around the `Framework` object. This means that if we put it into its own package it would have to import the `framwork` package and not be a useful refactor at all.

So it seems like I'll just have to see what random methods within that file can be moved without needing `framework`. Was that your assessment as well?

**Pull request closed without merging**